ls1intum/Artemis
# Feature/prevent force push

### Checklist

### Motivation and Context
Students should not be able to alter code, therefore force pushing should be blocked.

### Description
A REST-Call to the Bitbucket-server is made that protects all branches from force-pushing and deletion.

### Steps for Testing

1. Start the exercise in Artemis
2. Log in Bitbucket **as bitbucket administrator** and check the branch permissions:
![image](https://user-images.githubusercontent.com/5084100/64452269-46c6cd00-d0e6-11e9-9e9a-cc8983e8d3e0.png)
3. As student, commit & push a change, amend the commit and try to force-push: it should be rejected.
